src/js/utils/in-interaction-event.ts

```diff
@@ -0,0 +1,6 @@
+import { GenericObject } from 'types/generic.type';
+
+export function inInteraction(event: GenericObject | undefined): boolean {
+    event = event || window.event;
+    return !!event && /^(?:mouse|pointer|touch|gesture|click|key)/.test(event.type);
```

Comment:
You can leave in the  ` | undefined` above to match the type of `window.event` (`Event | undefined`) but it would be better to keep the function definition strict (optional `Event`) and conform to `window.event` here with a new const.

 ```suggestion
    const interaction: Event | undefined = event || window.event;
    return !!interaction && /^(?:mouse|pointer|touch|gesture|click|key)/.test(interaction.type);
```

Reply:
We're re-assigning so type is already enforced

![image](https://user-images.githubusercontent.com/2903955/75372098-ed21ad00-5895-11ea-90e6-02910e01a4ed.png)